src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoQueryModel.cs

```diff
@@ -882,6 +1030,27 @@ public override  IEnumerable<SqlSmoObject> Query(SmoQueryContext context, string
                     return ret;
                 }
             }
+            Schema parentSchema = context.Parent as Schema;
```

Comment:
are all these duplicated snippets code-gened?  If not if looks like it could be a template function pretty simply.